backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/optimizations/EscapeAnalysis.kt

```diff
@@ -1661,13 +1664,27 @@ internal object EscapeAnalysis {
                          callGraph: CallGraph, lifetimes: MutableMap<IrElement, Lifetime>) {
         assert(lifetimes.isEmpty())
 
-        val intraproceduralAnalysisResult =
-                IntraproceduralAnalysis(context, moduleDFG, externalModulesDFG, callGraph).analyze()
-        InterproceduralAnalysis(context, callGraph, intraproceduralAnalysisResult, externalModulesDFG, lifetimes,
-                // TODO: This is a bit conservative, but for more aggressive option some support from runtime is
-                // needed (namely, determining that a pointer is from the stack; this is easy for x86 or x64,
-                //         but what about all other platforms?).
-                propagateExiledToHeapObjects = true
-        ).analyze()
+        try {
+            val intraproceduralAnalysisResult =
+                    IntraproceduralAnalysis(context, moduleDFG, externalModulesDFG, callGraph).analyze()
+            InterproceduralAnalysis(context, callGraph, intraproceduralAnalysisResult, externalModulesDFG, lifetimes,
+                    // TODO: This is a bit conservative, but for more aggressive option some support from runtime is
+                    // needed (namely, determining that a pointer is from the stack; this is easy for x86 or x64,
+                    //         but what about all other platforms?).
+                    propagateExiledToHeapObjects = true
+            ).analyze()
+        } catch (t: Throwable) {
+            val extraUserInfo =
+                        """
+                        Please try to disable escape analysis and rerun the build. To do it add "-Xdisable-phases=EscapeAnalysis" to the compilation command line or for Gradle, as an example:
+
+                            macosX64("macos") {
+                                compilations["main"].kotlinOptions.freeCompilerArgs += "-Xdisable-phases=EscapeAnalysis"
+                            }
```

Comment:
This advice is somewhat hard to apply in practice: a user has to infuse the code snippet right into a middle of a build script, and there are a lot of things that can go wrong when doing this.

@ilmat192 please help to make it (both snippet and the description above) more user-friendly.
I personally prefer this approach: https://kotlinlang.org/docs/reference/native/ios_symbolication.html#producing-dsym-for-release-kotlin-binaries
but it has at least one drawback -- it has to be different from Groovy and Kotlin DSLs.